### PR TITLE
Fix extra preparations and cleanups in GetPinnedFileIdsBench

### PR DESCRIPTION
**Pull request by dbw9580**
### What changes are proposed in this pull request?

Fix extra preparations and cleanups in GetPinnedFileIdsBench.

### Why are the changes needed?

`GetPinnedFileIdsBench` does not properly handle cases where the benchmark is running in a local process or in a job worker, causing unwanted extra runs of `prepare()` and `cleanup()`. This may cause the expected number of test files to be incorrect and fail the benchmark.

### Does this PR introduce any user facing changes?

No.

### CODE REVIEW

File: stress/shell/src/main/java/alluxio/stress/cli/GetPinnedFileIdsBench.java [hunk 1]
```diff
@@ -86,6 +86,12 @@ public void prepare() throws Exception {
     // So including that in the log can help associate the log to the run
     LOG.info("Task ID is {}", mBaseParameters.mId);
 
+    // the preparation is done by the invoking client
+    // so skip preparation when running in job worker
+    if (mBaseParameters.mDistributed) {
+      return;
```
Comment (jiacheliu3) [inline stress/shell/src/main/java/alluxio/stress/cli/GetPinnedFileIdsBench.java:92]: You may want to add some logs so it's easier to track what happened and what not

File: stress/shell/src/main/java/alluxio/stress/cli/GetPinnedFileIdsBench.java [hunk 2]
```diff
@@ -126,13 +132,16 @@ public void prepare() throws Exception {
...
-    AlluxioURI baseUri = new AlluxioURI(mParameters.mBasePath);
-    try (CloseableResource<alluxio.client.file.FileSystemMasterClient> client =
-             mFileSystemContext.acquireMasterClientResource()) {
-      LOG.info("Deleting test directory {}", baseUri);
-      client.get().delete(baseUri, DeletePOptions.newBuilder().setRecursive(true).build());
-    } catch (AlluxioStatusException e) {
-      LOG.warn("Failed to delete test directory {}, manual cleanup needed", baseUri, e);
+    // skip cleanup in job worker as the test files are to be cleaned up by the client
+    if (!mBaseParameters.mDistributed) {
+      AlluxioURI baseUri = new AlluxioURI(mParameters.mBasePath);
```
Comment (jiacheliu3) [inline stress/shell/src/main/java/alluxio/stress/cli/GetPinnedFileIdsBench.java:137]: Same here on the logging

**Review by jiacheliu3 — APPROVED** [review]
LGTM, thanks for the fix!

File: stress/shell/src/main/java/alluxio/stress/cli/GetPinnedFileIdsBench.java (hunk 1 above)
Comment (dbw9580) [reply stress/shell/src/main/java/alluxio/stress/cli/GetPinnedFileIdsBench.java:93]: Added logs

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**